libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/roadobject/reststop/RestStopFromIntersection.kt

```diff
@@ -0,0 +1,127 @@
+package com.mapbox.navigation.core.trip.roadobject.reststop
+
+import com.mapbox.api.directions.v5.models.Amenity
+import com.mapbox.api.directions.v5.models.RestStop
+import com.mapbox.api.directions.v5.models.StepIntersection
+import com.mapbox.geojson.Point
+
+/**
+ * The class provides access to [RestStop] available in [StepIntersection].
+ *
+ * @param location location of the rest stop
+ * @param name name of the rest stop
+ * @param type type of the rest stop
+ * @param amenities facilities available in the rest stop. See [Amenity]
+ */
+class RestStopFromIntersection private constructor(
+    val location: Point?,
```

Comment:
`StrepIntersection` location is NonNull, should be good to set it as well
```suggestion
    val location: Point,
```